Assignment 2.Rmd

```diff
@@ -19,82 +29,198 @@ First we will isolate the variables that are of interest: tid and stid
-#Add a variable that describes that they are teachers
+#makes the group shows as teachers
 V.TCH$group <- "teacher"
 
-#Now repeat this process for the students
+#follow the steps what teachers did
 V.STD <- select(D1, stid, s.gender, s.major)
 V.STD <- unique(V.STD)
 V.STD$group <- "student"
 
-#Make sure that the student and teacher data frames have the same variables names
+#check same variables names for both teachers and students
 names(V.TCH) <- c("id", "gender", "topic", "group")
 names(V.STD) <- c("id", "gender", "topic", "group")
 
-#Bind the two data frames together (you will get a warning because the teacher data frame has 5 types of id (A,B,C,D,E) and the student has 25 (1-30), this isn't a problem)
+#bind teahcers and students' data together
 VERTEX <- bind_rows(V.TCH, V.STD)
 
 ```
 
 Now we have both a Vertex and Edge list it is time to plot our graph!
 
 ```{r}
-#Load the igraph package
+#install igraph package
 
 library(igraph)
 
-#First we will make an object that contains the graph information using our two dataframes EDGE and VERTEX. Notice that we have made "directed = TRUE" - our graph is directed since discipline is being given from a teacher to a student.
+#teachers to studnents always have directions，so we can get a graph include all informations
 
 g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
 
-#Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
-
 plot(g,layout=layout.fruchterman.reingold)
 
-#There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
+#Now we have to color the nodes based on gender.
 
-plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+gender_colours =ifelse(VERTEX$gender == "female", yes="blue",no = "red")
 
-#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+VERTEX = cbind(VERTEX,gender_colours)
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender_colours)
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender_colours, edge.width=EDGE$count)
 
-plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
 ````
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+
+#when we get teachers and students total number, it helps we count
+
+s_total<- EDGE %>% group_by(to) %>% summarise(sum(count))
+names(s_total)<-c("id","count")
+t_total<-EDGE %>% group_by(from) %>% summarise(sum(count))
+names(t_total)<-c("id","count")
+
+VERTEX_new<-bind_rows(s_total,t_total)
```

Comment:
Make sure you are binding rows of the same type (EG - `factor` to `factor`, not `factor` to `character`).